### PR TITLE
feat(timm_crossvit): add timm CrossViT image-classification family

### PR DESCRIPTION
**Pull request by zhenshanx-nv**
## Background

TensorRT-Model-Connect has no CrossViT family. CrossViT is listed by
wang-xinyu/tensorrtx and does not fit `timm_vit`: it runs two vision
transformers side by side on the same image at two patch sizes, and they
exchange information through a cross-attention path that has no analogue in a
single-branch ViT.

## Exit Criteria

- `timm/crossvit_9_240.in1k` builds and classifies through the
  `timm_crossvit_image_classification` runtime strategy.
- Logits match the timm reference on the same weights and the same input.
- The family is registered in the shared matrices, workloads, perf release, and
  website tables, and the existing static encapsulation tests still pass.

Non-goals: quantized builds and tensor-parallel builds both raise
`NotImplementedError`; only `crossvit_9_240` is registered, and the
`crop_scale=True` variants are not handled.

## Implementation

New `timm_crossvit` family, following the layout the other timm families use: a
Python builder plugin that emits the network through the TensorRT Network API,
a C++ runtime model with its own image-preprocess seam, and an e2e model
directory with one manifest.

The two branches see the same image at different resolutions, so the smaller
branch's rescale happens **inside the engine** rather than in preprocessing:
the bundle has one `pixel_values` input, and each branch derives its own size
from its weights. `add_bicubic_resize` sets both the cubic coefficient
(`-0.75`) and `HALF_PIXEL` coordinates explicitly, because TensorRT and PyTorch
choose different defaults and a mismatch shifts every sampled pixel slightly
instead of failing.

The fusion path is the part worth reading. Only class tokens cross between
branches: each branch projects its class token into the other branch's width,
that single token attends over the other branch's patch tokens, and the result
is projected back and re-attached to its own patches. `_cross_attention` has
exactly one query, so its output is one token wide — that asymmetry is what
makes it different from the self-attention in `_block`.

Nearly all geometry is read from the checkpoint, including each branch's input
size: the positional table has one entry per patch plus the class token, and
the patches tile a square, so the size is `sqrt(tokens - 1) * patch`. The head
count is the one thing that cannot be recovered — every projection is square,
so the head split leaves no trace in any tensor shape — and it is stated as a
constant with that reason.

### Change categories

- [x] Model or runtime behavior
- [ ] Public API
- [ ] ABI
- [ ] Bundle or artifact format
- [ ] Dependencies
- [x] Documentation only
- [ ] CI or developer tooling

### Commands and Results

Unit and tooling suite:

```
python -m pytest tests/builder/ tests/tools/ tests/e2e_harness/ -q -n 4 \
  --dist=worksteal --import-mode=importlib -p no:cacheprovider --tb=line
3 failed, 4042 passed, 8 skipped, 23 warnings in 253.82s
```

The 3 failures are pre-existing on this machine and unrelated to this change:
two `test_model_proof_runner.py` cases call `/bin/cp --reflink` on a filesystem
that does not support cloning, and `TestAddGeluFcMlp::test_silu_activation`
fails the same way on an unmodified checkout.

Numerical parity, TRT fp32 against timm on identical weights and one fixed
random 1x3x240x240 input:

```
timm reference: missing=0 unexpected=0
max|diff| = 0.002020
corr      = 0.99999935
argmax    TRT=138 timm=138
top5 overlap = 5/5
RESULT: PASS
```

This also covers the in-engine bicubic rescale: the fine branch runs at 240 and
the coarse branch at 224, so a resize that did not match
`torch.nn.functional.interpolate` would show up here.

### Hardware, Environment, and Revisions

| Item | Value |
| --- | --- |
| Repository head | `92db1113` |
| Checkpoint | `timm/crossvit_9_240.in1k` @ `df3051d20d8addd16d03c4230046128c17e32826` |
| Reference | timm 1.0.28 |
| GPU | NVIDIA Blackwell (`2u2g-gen-0295`) |
| Container | project `Dockerfile.dev.x86`, Ubuntu 24.04, Python 3.12 |
| Precision | parity run fp32; manifest builds fp16 |

### Not Run / Remaining Gaps

- The full e2e harness case was not run locally; it needs the CI image cache.
  It is covered by the internal gate.
- Only fp32 was compared against the reference. The fp16 path is exercised by
  the manifest but not separately compared.
- Only `crossvit_9_240` was tested. The `crossvit_15_*` and `crossvit_18_*`
  variants share the discovery path but were not built.
- The `_240` checkpoints rescale rather than crop. A `crop_scale=True`
  checkpoint would need a centre crop instead, which this builder does not do.

## Notes For Future Readers

Suggested review order: the fusion loop in `build_engine`, then
`_cross_attention`; the registration diffs are mechanical and mirror the
previous timm families.

The branch indexing in the fusion loop is the easy thing to get wrong: branch
`i`'s projected class token is fused against branch `(i + 1) % n`'s patches and
so runs at *that* branch's width, then is projected back to its own. Every
combination of those indices produces correctly shaped tensors, so only the
numerical comparison catches a mistake.

### Risk level

- [x] Low
- [ ] Medium
- [ ] High

The change is additive. It creates a new family directory and inserts adjacent
lines in the shared registration files; no existing family shares code with it.

### CODE REVIEW

**Comment by zhenshanx-nv** [comment]
Superseded by #1217, which implements this family in the post-cutover layout introduced by #1188. This branch targets `python/tensorrt_model_connect/families/`, which no longer exists, so it cannot be rebased.
